### ISSUE TITLE
Add option to provide a template

### ISSUE THREAD

**Issue by LaurentGoderre** [theoephraim/grunt-i18n-gspreadsheet#3]
For one our our use-case we need to replace only some of the strings and place them in a template. Is this something you would be open too?

/cc @nschonni

**Comment by theoephraim** [comment]
Sorry I don't follow. Do you mean you'd like it to do some find/replace in
template files?

**Comment by nschonni** [comment]
You can check out our current task here https://github.com/wet-boew/wet-boew/blob/v4.0/tasks/i18n.js

**Comment by theoephraim** [comment]
Its a bit hard to follow your example without knowing the format of the spreadsheet and desired file output.

Since using locale files with a i18n plugin is quite common with express and all the templating languages, I think it makes sense to keep this as is, just spitting out the locale files to be ingested by something else.

I would recommend still using a custom task, but using https://github.com/theoephraim/node-google-spreadsheets to handle the spreadsheet connection. You can also take a look at https://github.com/theoephraim/grunt-i18n-gspreadsheet/blob/master/tasks/i18n_gspreadsheet.js to see how this package uses it.
Or you could use this plugin to generate the locale files and use those directly (its just a json file with a dictionary of translations).

**Comment by nschonni** [comment]
> Its a bit hard to follow your example without knowing the format of the spreadsheet and desired file output.

Whoops, https://github.com/wet-boew/wet-boew/blob/v4.0/src/i18n/i18n.csv, which is actually just a local copy (right now) of a google sheet, and https://github.com/wet-boew/wet-boew/blob/v4.0/src/i18n/base.js is the template that is filled out for each language.

**Comment by nschonni** [comment]
I'll have to look at your code closer because I was thinking that a direct JSON conversion would probably be better

**Comment by theoephraim** [comment]
It looks to me like you're overcomplicating things a bit.
You can generate the json you need directly from the spreadsheet without having to keep so many pieces in sync.

If you absolutely need those nested elements in your JSON output, you'll could do some post processing on the JSON file - or maybe we can figure out a key structure within the spreadsheet that could work, although currently the keys are totally flexible and I'd like to keep it that way if possible.

**Comment by LaurentGoderre** [comment]
IN this case though we don`t want all keys output anyways.
